### PR TITLE
refactor(backend): rename invoice -> incoming payment

### LINKED ISSUE #244: Rename `invoice` in `incomingPayment`

### PR DESCRIPTION
**Pull request by sabinebertram**
## Changes proposed in this pull request

- remove the term `invoice` in favor for `incoming_payment`
- compliant with [Open Payments v2](https://github.com/interledger/open-payments/tree/ahb-opv2/notes)

**Notes**: 
- ~~I have not renamed files yet to make changes visible. I will update that with another commit as soon as people have done a first review.~~
- The term `invoice` is also used by the [Pay library](https://github.com/interledgerjs/interledgerjs/tree/master/packages/pay), hence the term is still used when using that library

## Context

- fixes #244 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

File: packages/backend/src/app.ts [hunk 1]
```diff
@@ -295,20 +303,22 @@ export class App {
...
+      'incomingPaymentService'
+    )
+    return incomingPaymentService
       .processNext()
       .catch((err) => {
-        this.logger.warn({ error: err.message }, 'processInvoice error')
+        this.logger.warn({ error: err.message }, 'processIncomingPayment error')
         return true
       })
       .then((hasMoreWork) => {
```
Comment (wilsonianb) [inline packages/backend/src/app.ts:316]: I don't know why these changes prompted those build errors (or why similar worker functions aren't a problem), but this seems to fix it:
```
      .catch((err: Error) => {
        this.logger.warn({ error: err.message }, 'processIncomingPayment error')
        return true
      })
      .then((hasMoreWork: boolean) => {
```

**Review by wilsonianb — COMMENTED** [review]
:+1: I'm ready for the file renames

**Comment by wilsonianb** [comment]
What do you think about renaming the directory `backend/src/open_payments/payment/incoming` and outgoing payments can subsequently be moved to `backend/src/open_payments/payment/outgoing`?

**Comment by sabinebertram** [comment]
> What do you think about renaming the directory backend/src/open_payments/payment/incoming and outgoing payments can subsequently be moved to backend/src/open_payments/payment/outgoing?

:+1: I'm all for shorter names.

**Review by wilsonianb — DISMISSED** [review]
:+1: 
There's a few comments with the old invoice path:
https://github.com/interledger/rafiki/blob/main/packages/backend/src/accounting/service.ts#L36
https://github.com/interledger/rafiki/blob/main/packages/backend/src/connector/core/rafiki.ts#L27

**Review by wilsonianb — APPROVED** [review]
